internal/jwtverify/token_verifier_jwt.go

```diff
@@ -58,6 +78,39 @@ type SubscribeTokenClaims struct {
 	jwt.StandardClaims
 }
 
+type jwksManager struct{ *jwks.Manager }
+
+func (j *jwksManager) verify(token *jwt.Token) error {
+	ctx := context.Background()
+	kid := token.Header().KeyID
+
+	key, err := j.Manager.FetchKey(ctx, kid)
+	if err != nil {
+		return err
+	}
+
+	if key.Kty != "RSA" || key.Use != "sig" {
```

Comment:
I think that in scope of Centrifugo we can move Use check to manager fetch method - i.e. never deal with `enc` use at all (and never cache keys with use other than `sig`).